bpo-46775: OSError should call winerror_to_errno unconditionally (#32179)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2022-03-30-02-36-25.bpo-46775.e3Oxqf.rst

Update the constructor of :exc:`OSError` to use winerror_to_errno()

> unconditionally. Patch by Dong-hee Na.

Reviewer: Since this isn't going in the Windows section, it's worth mentioning that somewhere in the text.

Probably also worth mentioning the user-visible impact, which will be that some (obscure) OSErrors may now be raised as a subclass, and the OSError.errno field may now be different from the winerror field.

Author: @zooba 
Thanks, Steve, I updated the NEWS.d. 
Please let me know the newly updated NEWS entry is well enough as you want.
If not, you can suggest changing the sentence (SInce I am not a good English writer ;))

Some Windows system error codes(>= 10000) are now mapped into

> the correct errno while :exc:`OSError` is raised as a subclass.

Reviewer: ```suggestion
the correct errno and may now raise a subclass of :exc:`OSError`.
```

Update the constructor of :exc:`OSError` to use winerror_to_errno()
unconditionally. Some :exc:`OSError` now be raised as a subclass and the
:attr:`OSError.errno` follows the mapping policy of winerror_to_errno().

> Patch by Dong-hee Na.

Reviewer 2: winerror_to_errno() is an internal function, but don't mention it but explains the change to end users. What I understand is that a Windows error is now correctly converted to an ``errno`` number for error numbers greater than or equal to 10000. For example, the socket error WSAEACCES is now converted to EACCES.

Maybe even give the list of errors code with their numbers:

```
     case WSAEINTR: 
     case WSAEBADF: 
     case WSAEACCES: 
     case WSAEFAULT: 
     case WSAEINVAL: 
     case WSAEMFILE: 
```

https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--9000-11999-

Author: Thanks Victor :)
